toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -237,14 +238,23 @@ func cleanUp(ic *ImageCustomizerParameters) error {
 func CustomizeImage(ctx context.Context, buildDir string, baseConfigPath string, config *imagecustomizerapi.Config, inputImageFile string,
 	rpmsSources []string, outputImageFile string, outputImageFormat string,
 	useBaseImageRpmRepos bool, packageSnapshotTime string,
-) error {
+) (err error) {
 	ctx, span := otel.GetTracerProvider().Tracer(OtelTracerName).Start(ctx, "customize_image")
 	span.SetAttributes(
 		attribute.String("output_image_format", string(outputImageFormat)),
 	)
-	defer span.End()
+	defer func() {
+		if err != nil {
+			span.RecordError(err)
```

Comment:
The error message could theoretically contain user data. For example, if a postCustomization script failed. So, I am not sure it is wise to record the error message in telemetry.